File: kafka/spring-cloud-stream/consumer/src/main/java/br/com/leonardoferreira/consumer/listener/CreatedUserListener.java
package br.com.leonardoferreira.consumer.listener;

import br.com.leonardoferreira.CreatedUser;
import br.com.leonardoferreira.consumer.channel.CreatedUserChannel;
import java.util.concurrent.atomic.AtomicInteger;
import lombok.extern.slf4j.Slf4j;
import org.springframework.cloud.stream.annotation.StreamListener;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class CreatedUserListener {

    public final static AtomicInteger counter = new AtomicInteger(0);

    @StreamListener(CreatedUserChannel.CREATED_USER_INPUT)
    public void listen(final CreatedUser createdUser) {
        log.info("M=listen, createdUser={}", createdUser);
        counter.incrementAndGet();
    }

}

File: kafka/spring-cloud-stream-integrated-spring-kafka/spring-cloud-stream/src/main/kotlin/br/com/leonardoferreira/listener/CreatedUserListener.kt
package br.com.leonardoferreira.listener

import br.com.leonardoferreira.CreatedUser
import br.com.leonardoferreira.binding.CreatedUserBinding
import org.slf4j.LoggerFactory
import org.springframework.cloud.stream.annotation.StreamListener
import org.springframework.messaging.Message
import org.springframework.stereotype.Component

@Component
class CreatedUserListener {

    private val log = LoggerFactory.getLogger(CreatedUserListener::class.java)

    @StreamListener(
        value = CreatedUserBinding.CREATED_USER_INPUT,
        condition = "new String(headers['x-user-origin'])=='web'"
    )
    fun listen(message: Message<CreatedUser>) {
        log.info(
            "M=listen, I=user created, payload={}, customHeader={}",
            message.payload,
            message.headers["x-user-origin"]
        )
    }

}
